sopel/bot.py

```diff
@@ -342,6 +344,37 @@ def post_setup(self) -> None:
 
     # plugins management
 
+    def store_plugin_handler(
```

Comment:
Maybe this is a nitpick, but I would call this:

```suggestion
    def set_plugin_handler(
```

That is all it does. And if the handler name already exists in `self._plugins`, the existing handler will be overwritten, which seems _possibly_ bad (even though other parts of the plugin system are supposed to make sure only one plugin with a given name gets loaded).